Sources/Build/BuildPlan.swift

```diff
@@ -2442,7 +2442,8 @@ public class BuildPlan: SPMBuildCore.BuildPlan {
     /// Extracts the artifacts  from an artifactsArchive
     private func parseArtifactsArchive(for target: BinaryTarget) throws -> [ExecutableInfo] {
         try self.externalExecutablesCache.memoize(key: target) {
-            return try target.parseArtifactArchives(for: self.buildParameters.triple, fileSystem: self.fileSystem)
+            let execInfos = try target.parseArtifactArchives(for: self.buildParameters.triple, fileSystem: self.fileSystem)
```

Comment:
This func is called by BuildPlan computeDependencies; the filtering of triples is added here to preserve the existing behavior of computeDependencies but lmk if we should return wo it.